third_party/terraform/data_sources/data_source_monitoring_notification_channel.go

```diff
@@ -0,0 +1,97 @@
+package google
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceMonitoringNotificationChannel() *schema.Resource {
+	dsSchema := datasourceSchemaFromResourceSchema(resourceMonitoringNotificationChannel().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "display_name", "type")
```

Comment:
I think project has to be added too, to allow users to specify it explicitly.

Reply:
yes, good suggestion, have added it